### PR TITLE
search: add user/org/global setting to include forks or archives by default

### PR DESCRIPTION
**Pull request by rvantonder**
Confirmed working manually. Looking into how to add a test for site config things.

### CODE REVIEW

**Comment by slimsag** [comment]
Can we do this via global/org/user settings instead? should be pretty easy and then individual users can configure it too

- add here: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3dd3d40004250a7bc63b779a39afa9ca3fb0770/-/blob/schema/settings.schema.json
- query like this: https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3dd3d40004250a7bc63b779a39afa9ca3fb0770/-/blob/cmd/frontend/graphqlbackend/settings_cascade.go#L174-181

**Comment by rvantonder** [comment]
Yeah! Can you elaborate how to query these values, i.e., what do I call to get the Go value? I am only finding references in ts code, and I can't seem to grok the linked example.

**Comment by rvantonder** [comment]
oh wait, I think I get it. the values are in
```
subject              *settingsSubject
```
So it's sufficient to query for the user, and then assume the settings are propagated down into this value right?

**Comment by rvantonder** [comment]
Nope, I don't follow, don't know how to get those values :-)

**Comment by rvantonder** [comment]
Looks like this is the way https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3dd3d40004250a7bc63b779a39afa9ca3fb0770/-/blob/cmd/frontend/graphqlbackend/search.go#L300-308

File: schema/settings.schema.json [hunk 1]
```diff
@@ -100,6 +100,18 @@
       "type": "string",
       "pattern": "literal|regexp"
     },
+    "search.includeForksEnabled": {
```
Comment (slimsag) [inline schema/settings.schema.json:103]: Thoughts on calling these `search.includeForks` and `search.includeArchived` instead? I'm not sure `Enabled` adds much

Comment (rvantonder) [reply schema/settings.schema.json:103]: Yeah

**Comment by rvantonder** [comment]
This addition breaks tests, because all sorts of things related to settings take on nil values. The first case is `ExtensionRegistry` [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d3dd3d40004250a7bc63b779a39afa9ca3fb0770/-/blob/cmd/frontend/graphqlbackend/default_settings.go#L68), but if that is fixed, other things crop up. I think I have to mock out all of `LatestSettings`, but don't have the time to do this now. Will look again tomorrow.

**Comment by slimsag** [comment]
Thanks for all your hard work on this! I pushed up a commit that mocks out `LatestSettings` and fixes the tests, PTAL as soon as you get a chance

**Comment by rvantonder** [comment]
@slimsag LGTM, I manually tested the setting in each of user/org/global settings. Ship it.

**Pull request merged**